discovery_engine_core/core/src/engine.rs

```diff
@@ -1065,230 +1114,290 @@ enum SearchBy<'a> {
+            let config = InitConfig {
+                api_key: "test-token".into(),
+                api_base_url: server.uri(),
+                markets: vec![Market::new("en", "US")],
+                // This triggers the trusted sources stack to also fetch articles
+                trusted_sources: vec!["example.com".into()],
+                excluded_sources: vec![],
+                smbert_vocab: format!("{asset_base}/smbert_v0001/vocab.txt"),
+                smbert_model: format!("{asset_base}/smbert_v0001/smbert-mocked.onnx"),
+                kpe_vocab: format!("{asset_base}/kpe_v0001/vocab.txt"),
+                kpe_model: format!("{asset_base}/kpe_v0001/bert-mocked.onnx"),
+                kpe_cnn: format!("{asset_base}/kpe_v0001/cnn.binparams"),
+                kpe_classifier: format!("{asset_base}/kpe_v0001/classifier.binparams"),
+                de_config: None,
+                log_file: None,
+                news_provider_path: "newscatcher/news-endpoint-name".into(),
+                headlines_provider_path: "newscatcher/headlines-endpoint-name".into(),
+            };
 
-        let asset_base = "../../discovery_engine_flutter/example/assets/";
-        let market = Market::new("en", "US");
-        let config = InitConfig {
-            api_key: "test-token".to_string(),
-            api_base_url: mock_server.uri(),
-            markets: vec![market.clone()],
-            // This triggers the trusted sources stack to also fetch articles
-            trusted_sources: vec!["example.com".to_string()],
-            excluded_sources: vec![],
-            smbert_vocab: format!("{}/smbert_v0001/vocab.txt", asset_base),
-            smbert_model: format!("{}/smbert_v0001/smbert-mocked.onnx", asset_base),
-            kpe_vocab: format!("{}/kpe_v0001/vocab.txt", asset_base),
-            kpe_model: format!("{}/kpe_v0001/bert-mocked.onnx", asset_base),
-            kpe_cnn: format!("{}/kpe_v0001/cnn.binparams", asset_base),
-            kpe_classifier: format!("{}/kpe_v0001/classifier.binparams", asset_base),
-            de_config: None,
-            log_file: None,
-            news_provider_path: "newscatcher/news-endpoint-name".to_string(),
-            headlines_provider_path: "newscatcher/headlines-endpoint-name".to_string(),
+            // Now we can initialize the engine with no previous history or state. This should
+            // be the same as when it's initialized for the first time after the app is downloaded.
+            let engine = Engine::from_config(config, None, &[], &[]).await.unwrap();
+
+            Mutex::new((server, engine))
         };
-        let endpoint_config = EndpointConfig::default()
-            .with_init_config(config.clone())
-            .await;
-        let providers = Arc::new(Providers::new(&config.clone().into()).unwrap());
+        let mut engine = MutexGuard::map(ENGINE.get_or_init(engine).await.lock().await, |engine| {
```

Comment:
:thinking: I'm not sure I like this "reuse engine but clear all state approach" compared to a "create anew but explicitly reuse some immutable parts approach" but I guess it's simpler here so I will approve it anyway.

Reply:
i'll try and see if i can change it into the suggested direction.

Reply:
my complain was mainly wrt. the engine as a whole then how we change the stacks, as it's prone to forgetting adding clearing for some data we might add in the future.

but like I said it's not too much of a problem so it's fine either way, I guess